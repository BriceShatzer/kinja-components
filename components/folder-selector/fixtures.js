// @flow

import React from 'react';
import BlogAvatar from '../blog-avatar';
import type {Item, ItemId, Level} from './types';

const level1Selection: ItemId = '1';
const level2Selection: ItemId = '2';
const level3Selection: ItemId = '1';

const blogItemGizmodo: Item = {name: 'Giz', id: level1Selection, icon: <BlogAvatar name="gizmodo" size={18} />};
const blogItemAVClub: Item = {name: 'AV', id: '2'};

const blogs = [
	{...blogItemGizmodo},
	{...blogItemAVClub}
];

const storyTypes = [
	{name: 'Some story type', id: '1'},
	{name: 'Some other story type', id: level2Selection}
];

const categories = [
	{name: 'Some category', id: level3Selection},
	{name: 'Some other category with a really long name', id: '2'}
];

const subcategories = [
	{name: 'Some subcategory', id: '1'},
	{name: 'Some other subcategory', id: '2'}
];

const levels: Array<Level> = [
	{
		id: 'blogs',
		displayName: 'Blogs',
		getItems: () => Promise.resolve(blogs),
		items: blogs,
		selection: level1Selection,
		required: true
	},
	{
		id: 'storyTypes',
		displayName: 'Story Types',
		getItems: () => Promise.resolve(storyTypes),
		items: storyTypes,
		selection: level2Selection
	},
	{
		id: 'categories',
		displayName: 'Categories',
		getItems: () => Promise.resolve(categories),
		items: categories,
		selection: level3Selection
	},
	{
		id: 'subcategories',
		displayName: 'Sub-categories',
		getItems: () => Promise.resolve(subcategories)
	}
];

// Multiple Selection
const blogsForMultipleSelection = [
	{ name: 'Gizmodo', id: '1', icon: <BlogAvatar name="gizmodo" size={18} />, hasChildren: true, selectedChildren: [], isLastSelection: true },
	{ name: 'Splinter', id: '2', icon: <BlogAvatar name="splinter" size={18} />, hasChildren: true, selectedChildren: [], isLastSelection: true },
	{ name: 'The Takeout', id: '3', icon: <BlogAvatar name="thetakeout" size={18} />, selectedChildren: [], isLastSelection: true }
];

const blogItemTheTakeout = {
	icon: <BlogAvatar name="thetakeout" size={18} />,
	id: '3',
	isLastSelection: true,
	name: 'The Takeout',
	selectedChildren: []
};

const Verticals = [
	[
		{ name: 'Earther', id: '11', parentId: '1' },
		{ name: 'io9', id: '12', parentId: '1' },
		{ name: 'Paleofuture', id: '13', parentId: '1' }
	],
	[
		{ name: 'Offspring', id: '21', parentId: '2' },
		{ name: 'Skillet', id: '22', parentId: '2' },
		{ name: 'Two Cents', id: '23', parentId: '2' },
		{ name: 'Vitals', id: '24', parentId: '2' }
	],
	[]
];

const levelsForMultipleSelection: Array<Level> = [
	{
		id: 'parentBlog',
		displayName: 'Parent Blog',
		getItems: () => Promise.resolve(blogsForMultipleSelection),
		items: blogsForMultipleSelection,
		required: false
	},
	{
		id: 'verticals',
		displayName: 'Verticals',
		getItems: id => id ? Promise.resolve(Verticals[parseInt(id) - 1]) : Promise.resolve([])
	}
];

export {
	blogItemGizmodo,
	blogItemAVClub,
	blogItemTheTakeout,
	blogsForMultipleSelection,
	levels,
	levelsForMultipleSelection
};
