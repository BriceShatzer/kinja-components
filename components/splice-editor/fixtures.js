/* @flow */

import React from 'react';
import Post from 'kinja-magma/models/Post';
import Blog from 'kinja-magma/models/Blog';
import BlogAvatar from '../blog-avatar';

import post from 'kinja-components/__stubs__/stubbedPostResponse.json';
import blog from 'kinja-components/__stubs__/gizmodo.json';


export const currentPost = Post.fromJSON({
	...post,
	...{
		headline: post.headline,
		properties: JSON.stringify({
			...JSON.parse(post.properties),
			...{
				isEmbiggened: true
			}
		})
	}
});

export const currentBlog = Blog.fromJSON({
	...blog,
	...{
		properties: {
			...blog.properties,
			...{
				blogGroup: 'gizmodo'
			}
		}
	}
});


export const selectedBlogs = [
	{
		icon: <BlogAvatar name="gizmodo" size={18} />,
		id: 12,
		name: 'io9',
		parentId: '1'
	},
	{
		icon: <BlogAvatar name="gizmodo" size={18} />,
		id: 13,
		name: 'Paleofuture',
		parentId: '1'
	},
	{
		icon: <BlogAvatar name="thetakeout" size={18} />,
		id: 3,
		name: 'The Takeout'
	},
	{
		icon: <BlogAvatar name="splinter" size={18} />,
		id: 2,
		name: 'Splinter'
	}
];


export const lastSharedBlogs = [
	{
		blogGroup: 'kotaku',
		createTimeMillis: 1552045361570,
		displayName: 'Cosplay',
		id: '1538697436',
		repostTimeMillis: 1553045361570
	},
	{
		blogGroup: 'gizmodo',
		displayName: 'Earther',
		createTimeMillis: 1551045261570,
		id: '1636390014',
		repostTimeMillis: 1553045261570
	},
	{
		blogGroup: 'splinter',
		displayName: 'Splinter',
		createTimeMillis: 1551045261570,
		id: '1635895473',
		repostTimeMillis: 1551245261570
	}
];