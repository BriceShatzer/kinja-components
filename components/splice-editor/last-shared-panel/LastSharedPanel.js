/* @flow */

import * as React from 'react';
import styled from 'styled-components';
import { DateTime } from 'luxon';

import { EnsureDefaultTheme } from 'kinja-components/components/theme';
import media from 'kinja-components/style-utils/media';
import UnshareModal from './UnshareModal';

// ICONS
import CloseIcon from '../../icon19/Close';
import ChevronUpIcon from '../../icon19/ChevronUp';
import ChevronDownIcon from '../../icon19/ChevronDown';
import BlogAvatar from '../../blog-avatar';

import Button, { ButtonWrapper } from 'kinja-components/components/buttons/Button';
import createTranslate from 'kinja-components/components/translator';
import translations from '../translations';

import type { BlogId } from 'kinja-magma/models/Id';
import type { LastSharedBlog } from '../';


const LastShare = styled.div`
	display: flex;
`;

const TitleText = styled.div`
	text-align: center;
	margin-bottom: 8px;
`;

const TitleTime = styled.div`
	display: flex;
	justify-content: center;
`;

export const Title = styled.div`
	display: flex;
	flex-direction: column;
	margin-bottom: 0;
	font-size: 20px;
	line-height: 22px;
	color: ${props => props.theme.color.black};
	cursor: pointer;

	${LastShare} {
		color: ${props => props.theme.color.primary};
	}
`;

const Offset = styled.div`
	font-size: 16px;
	line-height: 18px;
	align-self: flex-end;
`;

const BlogContainer = styled.div`
	width: 100%;
	padding-top: 30px;
`;

const IconOnDesktop = styled.div`
	margin-right: 10px;
	margin-bottom: 2px;
`;

const IconOnMobile = styled.div`
	position: relative;
	top: -1px;
	margin-right: 10px;
	cursor: pointer;
`;

export const BlogItem = styled.div`
	display: flex;
	align-items: center;
	height: 34px;

	&:not(:last-child) {
		margin-bottom: 10px;
	}

	${IconOnDesktop} {
		display: none;
	}

	${ButtonWrapper} {
		display: none;
		position: relative;
		top: 4px;
	}
`;

const BlogName = styled.span`
	width: 160px;
	font-size: 16px;
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
`;

const BlogSharedTime = styled.div`
	flex: 1;
	line-height: 19px;
	user-select: none;

	${LastShare} {
		justify-content: flex-end;
	}

	${Offset} {
		font-size: 14px;
	}
`;

export const Container = styled.div`
	width: 100%;
	text-align: left;

	${media.mediumUp`
		${Title} {
			flex-direction: row;
		}

		${TitleText} {
			margin: 0;
		}

		${BlogItem} {
			align-items: flex-end;

			&:hover {
				${ButtonWrapper} {
					display: block;
				}
			}
		}

		${BlogName} {
			font-size: 20px;
		}

		${BlogSharedTime} {
			align-self: flex-end;
		}

		${IconOnMobile} {
			display: none;
		}

		${IconOnDesktop} {
			display: block;
		}

		${LastShare} {
			justify-content: flex-start;
		}

		${BlogSharedTime} {
			flex: unset;
			margin-right: 60px;

			${LastShare} {
				width: 180px;
			}
		}
	`}
`;


type Props = {
	locale: string,
	lastSharedBlogs: Array<LastSharedBlog>,
	onUnshare: (blogId: BlogId, displayName: string) => void,
	timezone: string
};

type State = {
	isOpen: boolean,
	lastSharedBlogs: Array<LastSharedBlog>,
	lastSharedMillis: number,
	showUnshareModal: boolean,
	unshareBlogDisplayName: ?string,
	unshareBlogId: ?BlogId
}

class LastSharedPanel extends React.Component<Props, State> {
	static defaultProps = {
		locale: 'en-US'
	}

	state = {
		isOpen: false,
		lastSharedBlogs: this.props.lastSharedBlogs,
		lastSharedMillis: this.getLastSharedMillis(this.props.lastSharedBlogs),
		showUnshareModal: false,
		unshareBlogDisplayName: '',
		unshareBlogId: null
	}

	getLastSharedMillis(lastSharedBlogs: Array<LastSharedBlog>) {
		return Math.max(...lastSharedBlogs.map(blog => blog.createTimeMillis));
	}

	getSharedTimeElement(timemillis: number) {
		const { timezone: zone } = this.props;
		const year = DateTime.fromMillis(timemillis, { zone }).toFormat('y');
		const month = DateTime.fromMillis(timemillis, { zone }).toFormat('L');
		const day = DateTime.fromMillis(timemillis, { zone }).toFormat('dd');
		const date = `${month}/${day}/${year}`;
		const hour = DateTime.fromMillis(timemillis, { zone }).toFormat('h');
		const minute = DateTime.fromMillis(timemillis, { zone }).toFormat('mm');
		const meridiem = DateTime.fromMillis(timemillis, { zone }).toFormat('a').toLowerCase();
		const offset = DateTime.fromMillis(timemillis, { zone }).toFormat('ZZZZ');

		return <LastShare>&#160;{date} {this.translate('at')} {hour}:{minute}{meridiem}&#160;<Offset>{offset}</Offset></LastShare>;
	}

	onTitleClick = () => {
		this.setState({ isOpen: !this.state.isOpen });
	}

	onUnshare = (id: BlogId, displayName: string) => {
		this.setState({
			showUnshareModal: true,
			unshareBlogDisplayName: displayName,
			unshareBlogId: id
		});
	}

	onUnshareConfirm = () => {
		if (this.state.unshareBlogDisplayName && this.state.unshareBlogId) {
			this.setState({
				lastSharedBlogs: this.state.lastSharedBlogs.filter(blog => blog.id !== this.state.unshareBlogId)
			}, () => {
				if (this.state.unshareBlogId && this.state.unshareBlogDisplayName) {
					this.props.onUnshare(this.state.unshareBlogId, this.state.unshareBlogDisplayName);
					this.onModalClose();
				}
			});
		}
	}

	onModalClose = () => {
		this.setState({
			showUnshareModal: false,
			unshareBlogDisplayName: '',
			unshareBlogId: null
		});
	}

	translate = createTranslate(translations, this.props.locale);
	render() {
		return (
			<EnsureDefaultTheme>
				<React.Fragment>
					<Container>
						<Title onClick={this.onTitleClick}>
							<TitleText>{this.translate('Was last shared on')}</TitleText>
							<TitleTime>
								{this.getSharedTimeElement(this.state.lastSharedMillis)}
								{this.state.isOpen ? <ChevronUpIcon /> : <ChevronDownIcon />}
							</TitleTime>
						</Title>
						{this.state.isOpen &&
							<BlogContainer>
								{this.state.lastSharedBlogs.map(blog => {
									return (
										<BlogItem key={blog.id}>
											<IconOnMobile onClick={() => this.onUnshare(blog.id, blog.displayName)}><CloseIcon /></IconOnMobile>
											{blog.blogGroup
												? <IconOnDesktop><BlogAvatar name={blog.blogGroup} size={20} /></IconOnDesktop>
												: <IconOnDesktop><BlogAvatar name="kinja" size={18} /></IconOnDesktop>
											}
											<BlogName>{blog.displayName}</BlogName>
											<BlogSharedTime>{this.getSharedTimeElement(blog.repostTimeMillis)}</BlogSharedTime>
											<Button weight="secondary" label="Unshare" small
												onClick={() => this.onUnshare(blog.id, blog.displayName)} />
										</BlogItem>
									);
								})}
							</BlogContainer>
						}
					</Container>

					{this.state.showUnshareModal && this.state.unshareBlogDisplayName &&
						<UnshareModal
							displayName={this.state.unshareBlogDisplayName}
							isOpen={this.state.showUnshareModal}
							locale={this.props.locale}
							onClose={this.onModalClose}
							onSubmit={this.onUnshareConfirm}
						/>
					}
				</React.Fragment>
			</EnsureDefaultTheme>
		);
	}
}

export default LastSharedPanel;