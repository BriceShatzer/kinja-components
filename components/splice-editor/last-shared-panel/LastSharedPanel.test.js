import * as React from 'react';
import { mount } from 'enzyme';

import LastSharedPanel, { BlogItem, Title } from './LastSharedPanel';
import { lastSharedBlogs } from '../fixtures';


describe('<LastSharedPanel />', () => {
	const onUnshare = jest.fn();
	const component = mount(<LastSharedPanel
		lastSharedBlogs={lastSharedBlogs}
		onUnshare={onUnshare}
		timezone="America/New_York"
	/>);

	it('should render properly', () => {
		expect(component).toMatchSnapshot();
	});

	it('should open the panel', () => {
		const title = component.find(Title);
		title.simulate('click');
		component.update();

		expect(component.state().isOpen).toBe(true);
	});

	it('should call onUnshare and update state', () => {
		const blogItem = component.find(BlogItem).at(0);
		blogItem.simulate('mouseover');

		const button = blogItem.find({ label: 'Unshare' });
		button.simulate('click');

		const unshareModal = component.find('UnshareModal');
		expect(unshareModal).toBeTruthy();

		const okay = unshareModal.find({ label: 'Okay' });
		okay.simulate('click');

		expect(onUnshare).toHaveBeenCalledTimes(1);
		expect(onUnshare).toHaveBeenCalledWith('1538697436', 'Cosplay');

		const lastSharedBlogs = component.state().lastSharedBlogs;
		expect(lastSharedBlogs).toHaveLength(2);
		onUnshare.mockClear();
	});
});