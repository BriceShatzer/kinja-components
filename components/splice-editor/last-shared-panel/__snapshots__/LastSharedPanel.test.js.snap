// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LastSharedPanel /> should render properly 1`] = `
.c9 {
  fill: currentColor;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  text-align: center;
  margin-bottom: 8px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 0;
  font-size: 20px;
  line-height: 22px;
  color: #000;
  cursor: pointer;
}

.c2 .c6 {
  color: #0A7BC2;
}

.c8 {
  font-size: 16px;
  line-height: 18px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c0 {
  width: 100%;
  text-align: left;
}

@media only screen and (min-width:37.37em) {
  .c0 .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .c3 {
    margin: 0;
  }

  .c0 .LastSharedPanel__BlogItem-o5gdo2-8 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }

  .c0 .LastSharedPanel__BlogItem-o5gdo2-8:hover .Button__ButtonWrapper-j48i5d-2 {
    display: block;
  }

  .c0 .LastSharedPanel__BlogName-o5gdo2-9 {
    font-size: 20px;
  }

  .c0 .LastSharedPanel__BlogSharedTime-o5gdo2-10 {
    -webkit-align-self: flex-end;
    -ms-flex-item-align: end;
    align-self: flex-end;
  }

  .c0 .LastSharedPanel__IconOnMobile-o5gdo2-7 {
    display: none;
  }

  .c0 .LastSharedPanel__IconOnDesktop-o5gdo2-6 {
    display: block;
  }

  .c0 .c6 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0 .LastSharedPanel__BlogSharedTime-o5gdo2-10 {
    -webkit-flex: unset;
    -ms-flex: unset;
    flex: unset;
    margin-right: 60px;
  }

  .c0 .LastSharedPanel__BlogSharedTime-o5gdo2-10 .c6 {
    width: 180px;
  }
}

<LastSharedPanel
  lastSharedBlogs={
    Array [
      Object {
        "blogGroup": "kotaku",
        "createTimeMillis": 1552045361570,
        "displayName": "Cosplay",
        "id": "1538697436",
        "repostTimeMillis": 1553045361570,
      },
      Object {
        "blogGroup": "gizmodo",
        "createTimeMillis": 1551045261570,
        "displayName": "Earther",
        "id": "1636390014",
        "repostTimeMillis": 1553045261570,
      },
      Object {
        "blogGroup": "splinter",
        "createTimeMillis": 1551045261570,
        "displayName": "Splinter",
        "id": "1635895473",
        "repostTimeMillis": 1551245261570,
      },
    ]
  }
  locale="en-US"
  onUnshare={[MockFunction]}
  timezone="America/New_York"
>
  <EnsureDefaultTheme>
    <ThemeProvider
      theme={[Function]}
    >
      <LastSharedPanel__Container>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "LastSharedPanel__Container-o5gdo2-11",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "width:100%;text-align:left;",
                  "@media only screen and ",
                  "(min-width: 37.37em)",
                  "{",
                  "
		",
                  ".c1",
                  " {
			flex-direction: row;
		}

		",
                  ".c3",
                  " {
			margin: 0;
		}

		",
                  ".LastSharedPanel__BlogItem-o5gdo2-8",
                  " {
			align-items: flex-end;

			&:hover {
				",
                  ".Button__ButtonWrapper-j48i5d-2",
                  " {
					display: block;
				}
			}
		}

		",
                  ".LastSharedPanel__BlogName-o5gdo2-9",
                  " {
			font-size: 20px;
		}

		",
                  ".LastSharedPanel__BlogSharedTime-o5gdo2-10",
                  " {
			align-self: flex-end;
		}

		",
                  ".LastSharedPanel__IconOnMobile-o5gdo2-7",
                  " {
			display: none;
		}

		",
                  ".LastSharedPanel__IconOnDesktop-o5gdo2-6",
                  " {
			display: block;
		}

		",
                  ".c6",
                  " {
			justify-content: flex-start;
		}

		",
                  ".LastSharedPanel__BlogSharedTime-o5gdo2-10",
                  " {
			flex: unset;
			margin-right: 60px;

			",
                  ".c6",
                  " {
				width: 180px;
			}
		}
	",
                  "}",
                ],
              },
              "displayName": "LastSharedPanel__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "LastSharedPanel__Container-o5gdo2-11",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <LastSharedPanel__Title
              onClick={[Function]}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "c1",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "display:flex;flex-direction:column;margin-bottom:0;font-size:20px;line-height:22px;color:",
                        [Function],
                        ";cursor:pointer;",
                        ".c6",
                        "{color:",
                        [Function],
                        ";}",
                      ],
                    },
                    "displayName": "LastSharedPanel__Title",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "c1",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onClick={[Function]}
              >
                <div
                  className="c1 c2"
                  onClick={[Function]}
                >
                  <LastSharedPanel__TitleText>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "c3",
                            "isStatic": true,
                            "lastClassName": "c4",
                            "rules": Array [
                              "text-align:center;margin-bottom:8px;",
                            ],
                          },
                          "displayName": "LastSharedPanel__TitleText",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "c3",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c3 c4"
                      >
                        Was last shared on
                      </div>
                    </StyledComponent>
                  </LastSharedPanel__TitleText>
                  <LastSharedPanel__TitleTime>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "LastSharedPanel__TitleTime-o5gdo2-2",
                            "isStatic": true,
                            "lastClassName": "c5",
                            "rules": Array [
                              "display:flex;justify-content:center;",
                            ],
                          },
                          "displayName": "LastSharedPanel__TitleTime",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "LastSharedPanel__TitleTime-o5gdo2-2",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c5"
                      >
                        <LastSharedPanel__LastShare>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "c6",
                                  "isStatic": true,
                                  "lastClassName": "c7",
                                  "rules": Array [
                                    "display:flex;",
                                  ],
                                },
                                "displayName": "LastSharedPanel__LastShare",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "c6",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c6 c7"
                            >
                               
                              3/08/2019
                               
                              at
                               
                              6
                              :
                              42
                              am
                               
                              <LastSharedPanel__Offset>
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "LastSharedPanel__Offset-o5gdo2-4",
                                        "isStatic": true,
                                        "lastClassName": "c8",
                                        "rules": Array [
                                          "font-size:16px;line-height:18px;align-self:flex-end;",
                                        ],
                                      },
                                      "displayName": "LastSharedPanel__Offset",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "LastSharedPanel__Offset-o5gdo2-4",
                                      "target": "div",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                >
                                  <div
                                    className="c8"
                                  >
                                    EST
                                  </div>
                                </StyledComponent>
                              </LastSharedPanel__Offset>
                            </div>
                          </StyledComponent>
                        </LastSharedPanel__LastShare>
                        <ChevronDownIcon>
                          <Icon19>
                            <icon19__IconWrapper>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "icon19__IconWrapper-iyvn34-0",
                                      "isStatic": true,
                                      "lastClassName": "c9",
                                      "rules": Array [
                                        "fill:currentColor;display:flex;",
                                      ],
                                    },
                                    "displayName": "icon19__IconWrapper",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "icon19__IconWrapper-iyvn34-0",
                                    "target": "span",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <span
                                  className="c9"
                                >
                                  <ChevronDown
                                    aria-label="ChevronDown icon"
                                    height="18"
                                    viewBox="0 0 18 18"
                                    width="18"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <svg
                                      aria-label="ChevronDown icon"
                                      height="18"
                                      viewBox="0 0 18 18"
                                      width="18"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M3.87 6.16a.5.5 0 0 0-.74.68l5.5 6c.2.21.54.21.74 0l5.5-6a.5.5 0 1 0-.74-.68L9 11.76l-5.13-5.6z"
                                        fillRule="evenodd"
                                      />
                                    </svg>
                                  </ChevronDown>
                                </span>
                              </StyledComponent>
                            </icon19__IconWrapper>
                          </Icon19>
                        </ChevronDownIcon>
                      </div>
                    </StyledComponent>
                  </LastSharedPanel__TitleTime>
                </div>
              </StyledComponent>
            </LastSharedPanel__Title>
          </div>
        </StyledComponent>
      </LastSharedPanel__Container>
    </ThemeProvider>
  </EnsureDefaultTheme>
</LastSharedPanel>
`;
